src/browser/api/service.ts

```diff
@@ -50,18 +51,20 @@ export module Service {
     export function registerService (identity: Identity, serviceName?: string): ServiceIdentity | false {
```

Comment:
Why are we returning a union type here? This is a hack to use Javascript style "weakly typed" functions. API implementations should NAK by throwing errors, not returning "false".